File: hackerrank/java/30-days-of-code/day-24-more-linked-lists.java
import java.io.*;
import java.util.*;


class Node{
	int data;
	Node next;
	Node(int d){
        data = d;
        next = null; }

}


class Solution


{

    public static Node removeDuplicates(Node head) {
        Node temp = head;
      Set < Integer > s = new HashSet < Integer > ();
      s.add(temp.data);
      Node current = temp.next;
      while (current != null) {
          if (s.contains(current.data)) {
              temp.next = current.next;} else {
              s.add(current.data);
              temp = current;}
          current = current.next;}
      return head; }

    public static  Node insert(Node head, int data)

File: hackerrank/python/30-days-of-code/day-24-more-linked-lists.py
def removeDuplicates(self, head):
        temp = head
        s = set()
        s.add(temp.data)
        current = temp.next
        while current:
            if current.data in s:
                temp.next = current.next
            else:
                s.add(current.data)
                temp = current
            current = current.next
        return head
